WordPress/Classes/Models/ReaderCard+CoreDataClass.swift

```diff
@@ -2,6 +2,28 @@ import Foundation
 import CoreData
 
 public class ReaderCard: NSManagedObject {
+    enum CardType {
+        case post
+        case topics
+        case unknown
+    }
+
+    var type: CardType {
```

Comment:
Interesting. Wouldn't it be simpler to just do 
`var type: CardType = .unknown`
and then:
```
switch remoteCard.type {
        case .post:
            post = ReaderPost.createOrReplace(fromRemotePost: remoteCard.post, for: nil, context: managedObjectContext)
            self.type = .post
        case .interests:
            topics = NSOrderedSet(array: remoteCard.interests?.map {
                ReaderTagTopic.createIfNeeded(from: $0, context: context)
            } ?? [])
            self.type = .topics
        default:
            break
        }
```

Reply:
Any reason why you prefer this implementation?

My main points with the current are:

1. I find it a little bit cleaner since the logic of the card type is detached from the parsing;
2. It's a little bit safer: `ReaderPost.createOrReplace` can return nil, in that case, `type` would be `.unknown`. With this implementation, it will be `.post` and it might lead to weird issues.

Anyway, no big feelings. Let me know what do you prefer.

Reply:
As discussed in DM, In my mind, the local type should represent the remote type

Reply:
@yaelirub oops, I just realized that this needs to be a computed property.

`type` is not persisted to the database, so what happens is that it's always returning `.unknown` (because it's set only at the moment the object is created).

We can keep that way or I can remove the `type` property and refactor `cellForRowAt` to be like this:

```swift
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        guard let card = cards?[indexPath.row] else {
            return UITableViewCell()
        }

        if let post = card.post {
            return cell(for: card.post!, at: indexPath)
        } else if !card.topicsArray.isEmpty {
            return cell(for: card.topicsArray)
        } else {
            return UITableViewCell()
        }
    }
```

Thoughts?

Reply:
Gotcha. Up to you